src/python/gem5/resources/suite.py

```diff
@@ -0,0 +1,123 @@
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution;
+# neither the name of the copyright holders nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from .workload import Workload
+
+"""
+struct suite: {
+    id: 'id',
+    resource_version: '1.0.0',
+    gem5_versions: ['21.0.0', '21.1.0'],
+    workloads: [
+        {
+            id: 'id',
+            resource_version: '1.0.0',
+            input_group: ['input_group_id', 'input_group_id2'],
+        },
+        {
+            id: 'id2',
+            resource_version: '1.0.0',
+            input_group: ['input_group_id', 'input_group_id3'],
+        }
+    ]
+
+}
+class suite:
+
+impl: iter, next(maybe) , len
+
+methods: list_input_groups, with_input_group
+"""
+
+from _m5 import core
+
+from typing import Dict, Any, List, Optional, Set
+
+
+class Suite:
+    def __init__(
+        self,
+        resource_id: str,
+        resource_version: str,
+        workloads: List[Dict[str, Any]],
+        clients: Optional[List] = None,
+        gem5_version: Optional[str] = core.gem5Version,
+    ) -> None:
+
+        self.resource_id = resource_id
+        self.resource_version = resource_version
+        self.workloads = workloads
+        self.workload_obj_list = []
+        self.clients = clients
+        self.gem5_version = gem5_version
+        self.index = 0
+
+    def _obtain_workloads_list(self) -> None:
+        for workload in self.workloads:
+            self.workload_obj_list.append(
+                Workload(
+                    workload["id"],
+                    resource_version=workload["resource_version"],
+                    clients=self.clients,
+                    gem5_version=self.gem5_version,
+                )
+            )
+
+    def __iter__(self) -> "Suite":
+        return self
+
+    def __next__(self) -> Workload:
+        try:
+            workload = self.workload_obj_list[self.index]
+        except IndexError:
+            raise StopIteration
+        self.index += 1
+        return workload
```

Comment:
I think the following would work:
```suggestion
        for workload in self.workload_obj_list:
            yield workload
```
I think that if you make `__iter__` a generator then you don't need `__next__`. I could be wrong, this is a bit confusing.

Reply:
Don't forget to address this. It's still valid even though github says "outdated"